accord-core/src/main/java/accord/local/RedundantStatus.java

```diff
@@ -80,6 +92,26 @@ public enum RedundantStatus
      */
     LOCALLY_REDUNDANT,
 
+    /**
+     * The relevant owned ranges are all shard-redundant AND (pre-bootstrap or stale), meaning any intersecting transaction
```

Comment:
Broadly speaking, yes. I am not sure we need all of the distinctions, I suspect we can collapse some (and have added some TODOs to revisit), but I don't have time this second to refine them all. Perhaps when we have no failing seeds we can look at some simplifications.

Reply:
The main thing is making sure we can distinguish between locally redundant, shard redundant and no longer owned, plus cases where this is completely the case or only partially the case. But in different cases we care about different facets, e.g. when cleaning up locally we care about fully shard redundant (plus variations thereof for lost ownership), but when we are executing we care about locally partially redundant (plus variations thereof for lost ownership), and when we are fetching data from remote replicas that have erased their state we care only about _partial_ local redundancy (which means the transaction must have executed locally if it executes, so if we don't know it we can invalidate it). This is further complicated by bootstrap, where the distinction between whether it is fully or partially pre-bootstrap (and fully or partially shard redundant) matter. 

I hope to formalise the GC state machine once we're otherwise broadly where we want to be.

I did look at encoding each dimension independently but the refactor looked too big and didn't clearly simplify things - and would no doubt have introduced new problems to diagnose.

Reply:
It occurs to me a good approach might be to create separate (but very similar) enums, for each purpose. I'll add a TODO and explore this in a later patch.